presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/aggregation/GeometryStateFactory.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.geospatial.aggregation;
+
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.facebook.presto.array.ObjectBigArray;
+import com.facebook.presto.spi.function.AccumulatorStateFactory;
+import com.facebook.presto.spi.function.GroupedAccumulatorState;
+import org.openjdk.jol.info.ClassLayout;
+
+public class GeometryStateFactory
+        implements AccumulatorStateFactory<GeometryState>
+{
+    private static final long OGC_GEOMETRY_BASE_INSTANCE_SIZE = ClassLayout.parseClass(OGCGeometry.class).instanceSize();
+
+    @Override
+    public GeometryState createSingleState()
+    {
+        return new SingleGeometryState();
+    }
+
+    @Override
+    public Class<? extends GeometryState> getSingleStateClass()
+    {
+        return SingleGeometryState.class;
+    }
+
+    @Override
+    public GeometryState createGroupedState()
+    {
+        return new GroupedGeometryState();
+    }
+
+    @Override
+    public Class<? extends GeometryState> getGroupedStateClass()
+    {
+        return GroupedGeometryState.class;
+    }
+
+    public static class GroupedGeometryState
+            implements GeometryState, GroupedAccumulatorState
+    {
+        private long groupId;
+        private ObjectBigArray<OGCGeometry> geometries = new ObjectBigArray<>();
+        private long size;
+
+        @Override
+        public OGCGeometry getGeometry()
+        {
+            return geometries.get(groupId);
+        }
+
+        @Override
+        public void setGeometry(OGCGeometry geometry)
+        {
+            OGCGeometry previousValue = this.geometries.get(groupId);
```

Comment:
The implementation assumes that `geometry != null` and `previousValue  != null`. Could you add `checkArgument(geometry != null, "geometry is null");` and `verify(previousValue != null, format("Existing geometry value for group %s is null", groupId));`?

Reply:
The `set` method on the collection and `getGeometryMemorySize` are both null-safe, and I think that's safe.